app/controllers/volunteers_controller.rb

```diff
@@ -303,21 +301,32 @@ def home
     @total_shifts_needing_cov = Log.needing_coverage(current_volunteer.region_ids, 7).length
 
     # To Do Pickup Reports
-    @to_do_reports = Log.picked_up_by(current_volunteer.id, false)
+    @to_do_reports = Log.get_to_do_pick_up_reports(current_volunteer.id)
+
+    @by_month = get_reports_by_month
+
+    @assignment_names = get_assignment_names
+
+    @volunteer_stats_presenter = VolunteerStatsPresenter.new(current_volunteer)
+
+    render :home
+  end
 
-    @by_month = {}
+  private
+  def get_reports_by_month
+    by_month = {}
     Log.picked_up_by(current_volunteer.id).each do |log|
       year_month = log.when.strftime('%Y-%m')
-      @by_month[year_month] = 0.0 if @by_month[year_month].nil?
-      @by_month[year_month] += log.summed_weight unless log.summed_weight.nil?
+      by_month[year_month] = 0.0 if by_month[year_month].nil?
```

Comment:
do we need to check `if by_month[year_month].nil? `?

If the `by_month` variable is created as an empty hash on line 317, then by the time it gets to line 320, won't `by_month[year_month].nil?` always be true?